docs/modules/spark-k8s/pages/usage-guide/configuration-environment-overrides.adoc

```diff
@@ -0,0 +1,132 @@
+        security.properties:
+          networkaddress.cache.ttl: "30"
+      replicas: 1
+----
+
+Just as for the `config`, it is possible to specify this at the role level as well:
+
+[source,yaml]
+----
+nodes:
+  configOverrides:
+    security.properties:
+      networkaddress.cache.ttl: "30"
+  roleGroups:
+    default:
+      replicas: 1
+----
+
+All override property values must be strings.
+
+The same applies to the `job`, `driver` and `executor` roles of the SparkApplication.
+
+=== The spark-env.sh file
+
+The `spark-env.sh` file is used to set environment variables.
+Usually, environment variables are configured in `envOverrides`, but `envOverrides` only allow to set static values.
+The values in `spark-env.sh` are evaluated by the shell.
+For instance, if a SAS token is stored in a Secret and should be used for the Spark History Server, this token could be first stored in an environment variable via `podOverrides` and then added to the `SPARK_HISTORY_OPTS`:
+
+[source,yaml]
+----
+podOverrides:
+  spec:
+    containers:
+    - name: spark-history
+      env:
+      - name: SAS_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: adls-spark-credentials
+            key: sas-token
+configOverrides:
+  spark-env.sh:
+    SPARK_HISTORY_OPTS: >-
+      $SPARK_HISTORY_OPTS
+      -Dspark.hadoop.fs.azure.sas.fixed.token.mystorageaccount.dfs.core.windows.net=$SAS_TOKEN
+----
+
+NOTE: The given properties are written to `spark-env.sh` in the form `export KEY="VALUE"`.
+Make sure to escape the value already in the specification.
+Be aware that some environment variables may already be set, so prepend or append a reference to them in the value, as it is done in the example.
+
+=== The security.properties file
+
+The `security.properties` file is used to configure JVM security properties.
+It is very seldom that users need to tweak any of these, but there is one use-case that stands out, and that users need to be aware of: the JVM DNS cache.
+
+The JVM manages its own cache of successfully resolved host names as well as a cache of host names that cannot be resolved.
+Some products of the Stackable platform are very sensible to the contents of these caches and their performance is heavily affected by them.
+As of version 3.4.0, Apache Spark may perform poorly if the positive cache is disabled.
+To cache resolved host names, and thus speeding up queries you can configure the TTL of entries in the positive cache like this:
+
+[source,yaml]
+----
+spec:
+  nodes:
+    configOverrides:
+      security.properties:
+        networkaddress.cache.ttl: "30"
+        networkaddress.cache.negative.ttl: "0"
+----
+
+NOTE: The operator configures DNS caching by default as shown in the example above.
+
+For details on the JVM security see https://docs.oracle.com/en/java/javase/11/security/java-security-overview1.html
+
+
+== Environment Variables
+
+Similarly, environment variables can be (over)written. For example per role group:
```

Comment:
Environment variables can also be set (unfortunately, perhaps, due to historical reasons: it's the only operator that does so. See [here](https://github.com/stackabletech/spark-k8s-operator/blob/main/rust/crd/src/lib.rs#L208)) at a top level that then apply to job/driver/executor pods. Is it worth mentioning here for the sake of completion?

Reply:
Yes, that's worth mentioning.

457bfc60d1d7ca2c2f6f1dbfea89e2cb22a2c61e

Reply:
Thanks!